File: RedSheet.DbEntities/Models/ProjectNegotiation.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("ProjectNegotiations", Schema = "dbo")]
	[Serializable]
	public partial class ProjectNegotiation  
	{
 
	[Required]
	public System.DateTime EventDate { get; set; } 
 
	[Required] 
	[MaxLength(400)]
	public string IntangiblePowerPlan { get; set; } 
 
	[MaxLength(500)]
	public string KnowAboutThem { get; set; } 
 
	[MaxLength(750)]
	public string KnownIssues { get; set; } 
 
	[Required] 
	[MaxLength(50)]
	public string Location { get; set; } 
 
	[DatabaseGenerated(DatabaseGeneratedOption.Identity)] 
	[Key]
	public int ProjectNegotiationId { get; set; } 
 
	[RelationshipTableAttribue("ProjectModules","dbo")]
	//Relationships
	public int ProjectModuleId { get; set; } 
 
	[ForeignKey("ProjectModuleId")]
	public ProjectModule ProjectModule { get; set; } 
	[InverseProperty("ProjectNegotiation")]
	public ICollection<TheirTeamMember> TheirTeamMembers { get; set; } 

	[InverseProperty("ProjectNegotiation")]
	public ICollection<TheirTeamCommunicationMode> TheirTeamCommunicationModes { get; set; } 

	[InverseProperty("ProjectNegotiation")]
	public ICollection<Target> Targets { get; set; } 


	public ProjectNegotiation()
	{
	this.TheirTeamMembers = new HashSet<TheirTeamMember>();

	this.TheirTeamCommunicationModes = new HashSet<TheirTeamCommunicationMode>();

	this.Targets = new HashSet<Target>();

    }

    }
}

File: redsheet-portal/src/app/database-models/project-negotiation.ts
import { required, maxLength, range, nested } from '@rx/annotations';
import { ProjectModule, TheirTeamMember, TheirTeamCommunicationMode, Target,  vProjectNegotiationRecord  } from './'
export class ProjectNegotiation {
    constructor(projectNegotiation?: ProjectNegotiation  | vProjectNegotiationRecord )  {
        let properties = [ "eventDate", "intangiblePowerPlan", "knowAboutThem", "knownIssues", "location", "projectNegotiationId", "projectModuleId", "theirTeamMembers", "theirTeamCommunicationModes", "targets",];
        for (let property of properties)
            if (projectNegotiation && projectNegotiation[property])
                this[property] = projectNegotiation[property];
    }
 
    @required()
	eventDate : Date =   undefined;
 
    @required()
    @maxLength(400)
	intangiblePowerPlan : string =   undefined;
 
    @maxLength(500)
	knowAboutThem : string =   undefined;
 
    @maxLength(750)
	knownIssues : string =   undefined;
 
    @required()
    @maxLength(50)
	location : string =   undefined;
 
	projectNegotiationId : number =   0 ;
 
    @range(0,2147483647)
	projectModuleId : number =   undefined;
	projectModule : ProjectModule  ;
	@nested(TheirTeamMember)
	theirTeamMembers: TheirTeamMember[];

	@nested(TheirTeamCommunicationMode)
	theirTeamCommunicationModes: TheirTeamCommunicationMode[];

	@nested(Target)
	targets: Target[];



}
